web/src/app/user/user.component.ts

```diff
@@ -57,4 +59,32 @@ export class UserComponent implements OnInit{
       console.log("Logout error", error);
     });
   }
+
+  getUser() {
+    this.userService.getUser(this.user.id)
+        .subscribe((data: FirebaseUserModel) => this.user = { 
+          billingAddress: data['billingAddress'],
+          email: data['email'],
+          firstName: data['firstName'],
+          id: data['id'],
+          lastName: data['lastName'],
+          nickname: data['nickname'],
+          shippingAddress : data['shippingAddress'],
+          provider: this.user.provider,
+          image: this.user.image,
+          name: this.user.name
+        });
+  }
+
+  updateUser() {
+    this.user.name = this.user.firstName + " " + this.user.lastName;
+    this.userService.updateUser(this.user)
+        .subscribe(res=> {
+         console.log(res);
+        })
+    
+  }
+
+  get diagnostic() { return JSON.stringify(this.user); }
```

Comment:
Is this method actually being used?

Reply:
This is used for testing purposes to check whether or not an object is successfully bound to the form.